src/displayapp/screens/settings/SettingSetDateTime.cpp

```diff
@@ -0,0 +1,57 @@
+#include "displayapp/screens/settings/SettingSetDateTime.h"
+#include "displayapp/screens/settings/SettingSetDate.h"
+#include "displayapp/screens/settings/SettingSetTime.h"
+#include <lvgl/lvgl.h>
+#include <hal/nrf_rtc.h>
+#include <nrf_log.h>
+#include "displayapp/DisplayApp.h"
+#include "displayapp/screens/Symbols.h"
+#include "displayapp/screens/ScreenList.h"
+#include "displayapp/screens/Label.h"
+#include "components/settings/Settings.h"
+#include "displayapp/InfiniTimeTheme.h"
+
+using namespace Pinetime::Applications::Screens;
+
+bool SettingSetDateTime::OnTouchEvent(Pinetime::Applications::TouchEvents event) {
+  return screens.OnTouchEvent(event);
+}
+
+SettingSetDateTime::SettingSetDateTime(Pinetime::Applications::DisplayApp* app,
+                                       Pinetime::Controllers::DateTime& dateTimeController,
+                                       Pinetime::Controllers::Settings& settingsController)
+  : Screen(app),
+    dateTimeController {dateTimeController},
+    settingsController {settingsController},
+    screens {app,
+             0,
+             {[this]() -> std::unique_ptr<Screen> {
+                return screenSetDate();
+              },
+              [this]() -> std::unique_ptr<Screen> {
+                return screenSetTime();
+              }},
+             Screens::ScreenListModes::UpDown} {
+}
+
+std::unique_ptr<Screen> SettingSetDateTime::screenSetDate() {
+
+  return std::make_unique<Screens::SettingSetDate>(0, 2, app, dateTimeController, *this);
+}
```

Comment:
I think you should be able to create the dotIndicator in here, which would be a more appropriate place for it I think. Then the screen numbers don't need to be passed to the screens.

Reply:
I tried it and it just crashes.

Reply:
Works fine for me.

```suggestion
std::unique_ptr<Screen> SettingSetDateTime::screenSetDate() {
  Widgets::DotIndicator dotIndicator(0, 2);
  dotIndicator.Create();
  return std::make_unique<Screens::SettingSetDate>(app, dateTimeController, *this);
}
```

Reply:
looks like i did it wrong, oops